File: solutions/r/bob/2/bob.R
bob <- function(input) {

if (input == toupper(input) &&  length(grep('^.*[A-Z]+.*[^\\?] *$',input,value=TRUE, perl=TRUE)) > 0) {
	return("Whoa, chill out!")
}

if (input == toupper(input) && length(grep('^.*[A-Z]+.*\\? *$',input,value=TRUE, perl=TRUE)) > 0) {
	return("Calm down, I know what I'm doing!")
}

if (length(grep('^.*\\? *$',input,value=TRUE)) > 0) {
	return ("Sure.")
}

if (length(grep('^[\\t\\r\\n ]*$',input,value=TRUE, perl=TRUE)) > 0) {
	return("Fine. Be that way!")
}

return("Whatever.")

}

File: solutions/csharp/bob/2/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Response(string statement)
    {

        var pattern = @"^.*\? *$"; // Question
        Match m = Regex.Match(statement, pattern);

        if (m.Success)
        {
            pattern = @"^.*[a-zA-Z]+.*$"; // Question
            m = Regex.Match(statement, pattern);
            if (m.Success && statement.ToUpper() == statement) return "Calm down, I know what I'm doing!";
            else return "Sure.";
        }

        pattern = @"^[\t\r\n ]*$";
        m = Regex.Match(statement, pattern);

        if (m.Success) return "Fine. Be that way!";

        pattern = @"^.*[a-zA-Z]+.*$"; // Question
        m = Regex.Match(statement, pattern);

        if (m.Success && statement.ToUpper() == statement) return "Whoa, chill out!";

        return "Whatever.";
    }
}